### PR TITLE
UCS/QUEUE/FRAGLIST: Fix checking for last element in fraglist queue

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Potential fix for #6738
Thanks @SeyedMir for identifying the issue!

### CODE REVIEW

File: src/ucs/datastruct/queue.h [hunk 1]
```diff
@@ -107,22 +116,22 @@ static inline void ucs_queue_del_iter(ucs_queue_head_t *queue, ucs_queue_iter_t
 {
     ucs_assert((iter != NULL) && (*iter != NULL));
 
-    if (queue->ptail == &(*iter)->next) {
+    if (ucs_queue_is_last(queue, *iter)) {
         queue->ptail = iter; /* deleting the last element */
```
Comment (hoopoepg) [inline src/ucs/datastruct/queue.h:120]: comment is really confusing: last could mean 'only one element in queue' and 'no next element in queue' :)

**Review by brminich — APPROVED** [review]

**Review by hoopoepg — APPROVED** [review]

**Review by hoopoepg — APPROVED** [review]

File: src/ucs/datastruct/queue.h (hunk 1 above)
Comment (yosefe) [reply src/ucs/datastruct/queue.h:120]: maybe ucs_queue_is_back() or ucs_queue_is_tail()?

Comment (brminich) [reply src/ucs/datastruct/queue.h:120]: `ucs_queue_is_tail()` sounds good to me

Comment (hoopoepg) [reply src/ucs/datastruct/queue.h:120]: yep, tail resolves all misunderstandings :)

**Review by brminich — APPROVED** [review]

**Review by hoopoepg — APPROVED** [review]

**Comment by yosefe** [comment]
bot:pipe:retest

File: src/ucs/datastruct/frag_list.c [hunk 2]
```diff
@@ -292,12 +292,15 @@ ucs_frag_list_insert_slow(ucs_frag_list_t *head, ucs_frag_list_elem_t *elem,
...
+            ucs_assertv(h->head.first_sn <= h->head.last_sn,
+                        "h=%p first_sn=%u last_sn=%u", h, h->head.first_sn,
+                        h->head.last_sn);
             /* add tail, check merge with next list */
             frag_list_add_tail(h, elem);
             nexth = ucs_container_of(h->list.next, ucs_frag_list_elem_t, list);
-
-            if (nexth != NULL && nexth->head.first_sn == sn + 1) {
+            if (!ucs_queue_is_tail(&head->list, &h->list) &&
+                (nexth->head.first_sn == sn + 1)) {
```
Comment (dmitrygx) [inline src/ucs/datastruct/frag_list.c:303]:
```suggestion
                (nexth->head.first_sn == (sn + 1))) {
```

**Review by dmitrygx — APPROVED** [review]

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**